WordPress/src/main/java/org/wordpress/android/util/WPMediaUtils.java

```diff
@@ -205,8 +202,19 @@ String getErrorMessage(final Context context, final MediaModel media, final Medi
                 return context.getString(R.string.error_media_parse_error);
             case GENERIC_ERROR:
                 return context.getString(R.string.error_generic_error);
+            case EXCEEDS_SITE_SPACE_QUOTA_LIMIT:
```

Comment:
I am not sure how I feel about adding these missing cases with just `null` values. I understand that technically there is no change. However, missing switch cases and specifically having them return `null` values have different meanings to me.

I am also not sure whether it makes sense to make it a part of this PR. I really like that you are fixing a bunch of things and I am all for it. I just think it's better to keep the PRs focused if possible (which I fail myself at times). A different approach might have been to have a really small PR that adds Glide to `EditPostSettingsFragment` and another one fixing these issues. In that case, I'd more confidently say that we should add proper error messages for the cases you just added.

Just my 2 cents.

Reply:
The goal here is to have all switch statements with explicitly specified cases. It all derives from `Fail ASAP principle` -> when a method expects just certain enum values, it should throw an exception for all the other values (or at least log an error message). We can't just start throwing an exception or logging error messages in a legacy code, since even missing cases might be expected, so explicitly specifying all the cases is all we can do. 
The biggest advantage is when we create a new enum value, lint won't let us build the project unless we add handling for the new enum value to all switch statements, which is very convenient.(we need to remove this switch statement from the lint-baseline obviously, but I do that from time to time not in each PR separately)

>I am also not sure whether it makes sense to make it a part of this PR.
This is a topic worth discussing with everyone on the team. 


I want to write a post about both these topics in a near future. Basically one of the reasons why I don't put small refactoring changes in separate PR is that Android Studio warns me about these issues when committing a change, so it's super convenient to fix it right away.

Reply:
> The goal here is to have all switch statements with explicitly specified cases.

Which I agree with you, but I don't see how it has anything to do with a PR titled as `Replace Volley with Glide`.

> The biggest advantage is when we create a new enum value, lint won't let us build the project unless we add handling for the new enum value to all switch statements, which is very convenient.

Again, I think I wasn't very clear in my comment. I have no arguments against having these missing cases. What I am worried about though is returning `null`. As I also tried to explain in my previous comment, having a `null` value for these cases means that we specifically want to return `null` whereas not having the cases at all means we just made a mistake by not updating this switch case.

I just feel that a better way to fix these issues is to open a separate PR where we return specific error messages for the missing cases. Hope that clears it up!

Reply:
>Which I agree with you, but I don't see how it has anything to do with a PR titled as Replace Volley with Glide.

I'll create a post about this topic. I feel like minor refactoring (without changing behavior) should be part of each commit/PR, even though it's not strictly related to the PR itself. 

>I just feel that a better way to fix these issues is to open a separate PR where we return specific error messages for the missing cases. Hope that clears it up!

I'm still not sure it's the way we should go. We'd change the behavior by returning specific error message and I don't think it's save to assume that all missing cases are errors -> they might be omitted on purpose, because the author wanted to return null. Explicitly returning null is definitely not ideal, but it's a change which doesn't change the behavior and moves us towards state, where it'll be safe to assume that lint takes care of warning us about all missing cases. 
It would definitely be better to take one switch statement at a time and do research whether we can safely add an error message or throw an exception. But this approach would take a lot of time and would be error prone with very little gain. That's the main reason why I decided to start fixing such switch statements by returning a default value.

Reply:
> I feel like minor refactoring (without changing behavior) should be part of each commit/PR, even though it's not strictly related to the PR itself.

I definitely agree with you. My concern was the change in the meaning and that it was not very relevant to the feature.

> We'd change the behavior by returning specific error message and I don't think it's save to assume that all missing cases are errors -> they might be omitted on purpose, because the author wanted to return null

My concern was exactly this, by adding those cases we'd just be changing the meaning of that code from `missing cases` to `these cases are specifically returning null`.

> It would definitely be better to take one switch statement at a time and do research whether we can safely add an error message or throw an exception.

If we are not willing to spend that time, we could at least write a comment in the code explaining why they are currently returning null.